src/main.rs

```diff
@@ -57,7 +78,10 @@ impl MyApp {
                         Err(_) => s.send(Message::ShowError("Error loading diff!".to_string())),
                     });
                 }
-                Message::UpdateAppData(app_data) => self.app_data = Some(app_data),
+                Message::UpdateAppData(app_data) => {
+                    self.app_data = Some(app_data);
+                    self.update_app_data();
+                }
```

Comment:
Pass the app_data from the Message directly to the function. That way you don't have to check inside of self.update_app_data() if it's not None.